client/secrets/rsa.go

```diff
@@ -0,0 +1,111 @@
+package secrets
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha512"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+)
+
+const (
+	DefaultEncryptionBits    = 2048
+	DefaultTypeRSAPublicKey  = "RSA PUBLIC KEY"
```

Comment:
I wouldn't call this "default"since it is not something you can configure.
Maybe:
```suggestion
	BlockTypeRSAPublicKey  = "RSA PUBLIC KEY"
```